scripts/verify-released-app-bundle.sh

```
@@ -0,0 +1,209 @@
+# self_test — build a synthetic bundle, ad-hoc sign it, prove run_checks accepts
+# it, then tamper a sealed Mach-O resource and prove run_checks rejects it with
+# the same failure class as issue #6670. Runs on macOS without any secrets.
+self_test() {
+  command -v /usr/bin/codesign >/dev/null || fail "self-test requires codesign (macOS)"
+  # Dynamic scoping: run_checks (called below) sees this, but the global stays
+  # clean for any other call path.
+  local REQUIRE_NOTARIZED=0
+
+  local workdir app
+  workdir="$(mktemp -d "${TMPDIR:-/tmp}/cmux-verify-selftest.XXXXXX")"
+  trap "rm -rf '$workdir'" EXIT
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)